lab_solutions/lab01_reduce.py

```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2020 Patrick Nanys <patrick.nanys2000@gmail.com>
+#
+# Distributed under terms of the MIT license.
+
+
+def reduce(l, acc_func, accumulator=None):
+    if accumulator is not None:
+        l.insert(0, accumulator)
```

Comment:
You should not modify `l`. It's passed as a reference so the original list is the one that's modified here.